### ISSUE TITLE
sRGB-related improvements

### ISSUE THREAD

**Issue by mosra** [mosra/magnum#125]
### sRGB/S3TC

After reading [The sRGB learning curve](https://gamedevdaily.io/the-srgb-learning-curve-773b7f68cf7a) (great article, by the way) I was investigating sRGB support in OpenGL and found out that the old [EXT_texture_sRGB](https://www.opengl.org/registry/specs/EXT/texture_sRGB.txt) extension (part of GL 2.1) had an interaction with [EXT_texture_compression_s3tc](https://www.opengl.org/registry/specs/EXT/texture_compression_s3tc.txt) and provided additional sRGB/DXT compression formats. It might be a good idea to provide these in the `TextureFormat` and `CompressedPixelFormat` enums:

        COMPRESSED_SRGB_S3TC_DXT1_EXT                  0x8C4C
        COMPRESSED_SRGB_ALPHA_S3TC_DXT1_EXT            0x8C4D
        COMPRESSED_SRGB_ALPHA_S3TC_DXT3_EXT            0x8C4E
        COMPRESSED_SRGB_ALPHA_S3TC_DXT5_EXT            0x8C4F

OpenGL ES is a bit different. The [EXT_sRGB](https://www.khronos.org/registry/gles/extensions/EXT/EXT_sRGB.txt) ES extension says the following and the S3TC/SRGB support is then added by [NV_sRGB_formats](https://www.khronos.org/registry/gles/extensions/NV/NV_sRGB_formats.txt) (ES2+).

    3) Should we allow the SRGB_*_S3TC_DXT* or any of the other COMPRESSED
    formats if the implementation supports any of those formats?
    
    No since all hardware doesn't support this.  It can be added as a separate
    extension if needed.

No such thing in WebGL, sadly.

- [ ] Add missing SRGB/S3TC formats

### S3TC/SRGB support in DDS importer

After quick googling and skimming through the `DdsImporter` plugin I wasn't able to find out whether the container supports these formats or not. (*Personally I'd expect all DXT-compressed images to be in sRGB (similarly as all PNG/JPEG images are implicitly in sRGB), but marking all imported images as sRGB is the same level of wrong as marking them all as RGB.*)

@Squareys, did you, by any chance, come across these formats when implementing the plugin? Do you know if there are any tools that provide RGB/sRGB option when exporting DXT-compressed images?

- [ ] Add SRGB/DXT support to DDS importer

### sRGB-related extensions on OpenGL ES

- [ ] [NV_generate_mipmap_sRGB](https://www.khronos.org/registry/gles/extensions/NV/NV_generate_mipmap_sRGB.txt) (ES2, I think ES3 doesn't have this restriction anymore)
- [ ] [EXT_texture_sRGB_R8](https://www.khronos.org/registry/gles/extensions/EXT/EXT_texture_sRGB_R8.txt), [EXT_texture_sRGB_RG8](https://www.khronos.org/registry/gles/extensions/EXT/EXT_texture_sRGB_RG8.txt) (ES3+, I guess something similar is/should be for desktops too)

**Comment by mosra** [comment]
Some *(scary!)* information about colorspaces in DDS files: http://stackoverflow.com/questions/13491131/what-are-the-exact-color-spaces-of-dxgi-formats

**Comment by mosra** [comment]
sRGB support in `Color[34]` classes done in 2b97e5336022841e1ff4d122d5acea4b10dc276f.

**Comment by mosra** [comment]
Just discovered that WebGL has [WEBGL_compressed_texture_s3tc_srgb](https://www.khronos.org/registry/webgl/extensions/WEBGL_compressed_texture_s3tc_srgb/).

**Comment by mosra** [comment]
Added http://www.ericbrasseur.org/gamma.html to the list, great set of test images. Chrome on my HiDPI screen is apparently *not* doing the right thing, it seems.

**Comment by Squareys** [comment]
Ha! Chome on Windows 10 without HiDPI fails the test also 😆

**Comment by mosra** [comment]
Nah, actually it seems to be doing the correct thing: http://www.ericbrasseur.org/gamma_dalai_lama.html

**Comment by mosra** [comment]
- sRGB variants of S3TC compression formats added in https://github.com/mosra/magnum/commit/02538d566f704324761cc587e989ac013121d4b6
- sRGB `PixelFormat` values and one and two-component sRGB texture formats added in https://github.com/mosra/magnum/commit/40b8815b9da1841b885bb2e7ac20aa8f0c3417c1
